static/js/src/advantage/subscribe/react/components/FreeTrialRadio.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { RadioInput, Row, Col } from "@canonical/react-components";
+import { formatter } from "../../renderers/form-renderer";
+import { add, format } from "date-fns";
+import usePreview from "../APICalls/usePreview";
+
+const DATE_FORMAT = "dd MMMM yyyy";
+
+type Props = {
+  isUsingFreeTrial: boolean;
+  setIsUsingFreeTrial: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const FreeTrialRadio = ({ isUsingFreeTrial, setIsUsingFreeTrial }: Props) => {
+  const { data: preview } = usePreview();
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setIsUsingFreeTrial(e.target.value === "useFreeTrial");
+  };
+
+  return (
+    <Row>
+      <Col size={11} emptyLarge={2}>
+        {isUsingFreeTrial && (
+          <>
+            <p>
+              <strong>Your free trial ends: </strong>
+              {format(
+                add(new Date(), {
+                  months: 1,
+                }),
+                DATE_FORMAT
+              )}{" "}
```

Comment:
There are some potential edge cases to consider as the end date calculation is handled entirely on the front end. I don't want to make a big deal out of it, just wanted to highlight - would be good to check if we're okay with this.

- What about users in different timezones 
- What about users with an incorrectly set date on their machine (very unlikely, I know)

The above could result in a slightly different date than actual trial end date? If someone really tried (and made a screenshot) this could - in theory - have some legal implications (false/misleading claims).

For that reason I wonder if this value should come from the back-end?

Reply:
Hey! I don't see the issue with users in different time zones what would happen then?
For users with an incorrect date on their machine then yes I guess we would display a wrong date here.
I doubt we risk too much in terms of legal implications as it's easy to doctor a screenshot anyway and I'd argue that if you change the date on your machine then you can't expect everything to behave correctly (Yes I'm a lawyer now :judge: )
@wgx @SirSamTumless What do you think?

Reply:
@ClementChaumel fine to go ahead like this - as you say, nobody will be surprised if they have their clock set incorrectly.